File: Projects/usaco/dualpal.java
/*
ID: rombin82
LANG: JAVA
TASK: dualpal
*/

import java.io.*;
import java.util.*;

public class dualpal {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(new File("dualpal.in"));
		PrintWriter stdout = new PrintWriter(new File("dualpal.out"));
		Runtime.getRuntime().addShutdownHook(new Thread(() -> stdout.flush()));
		int N = in.nextInt();
		int S = in.nextInt();
		while (N-- > 0) {
			boolean flag = true;
			while (flag) {
				int num = ++S;
				if (isDualPal(num)) {
					stdout.println(num);
					flag = false;
					break;
				}
			}
		}
	}
	private static boolean isDualPal(int num) {
		int count = 0;
		for (int b = 2; b <= 10; b++)
			if (isPalindrome(Integer.toString(num, b)))
				count++;
		return count >= 2;
	}
	private static boolean isPalindrome(String str) {
		return new StringBuffer(str).reverse().toString().equals(str);
	}
}

File: Projects/usaco/dualpal.kt
/*
ID: rombin82
LANG: JAVA
TASK: dualpal
*/

import java.io.*
import java.util.*

object dualpal {
    @Throws(IOException::class)
    fun main(args: Array<String>) {
        val stdin = Scanner(File("dualpal.in"))
        val stdout = PrintWriter(File("dualpal.out"))
        Runtime.getRuntime().addShutdownHook(Thread { stdout.flush() })
        var N = stdin.nextInt()
        var S = stdin.nextInt()
        while (N-- > 0) {
            var flag = true
            while (flag) {
                val num = ++S
                if (isDualPal(num)) {
                    stdout.println(num)
                    flag = false
                    break
                }
            }
        }
    }

    private fun isDualPal(num: Int): Boolean {
        var count = 0
        for (b in 2..10)
            if (isPalindrome(Integer.toString(num, b)))
                count++
        return count >= 2
    }

    private fun isPalindrome(str: String): Boolean {
        return StringBuffer(str).reverse().toString().equals(str)
    }
}
